app/screens/post_options/bindings/index.ts

```diff
@@ -24,9 +25,22 @@ type OwnProps = {
     post: Post;
 }
 
+const getAppsBindings = makeAppBindingsSelector(AppBindingLocations.POST_MENU_ITEM);
+const getThreadAppsBindings = makeRHSAppBindingSelector(AppBindingLocations.POST_MENU_ITEM);
+
 function mapStateToProps(state: GlobalState, props: OwnProps) {
     const apps = appsEnabled(state);
-    const bindings = apps ? getAppsBindings(state, AppBindingLocations.POST_MENU_ITEM) : [];
+    let bindings: AppBinding[] | null = [];
+    if (apps) {
+        if (props.post.channel_id === getCurrentChannelId(state)) {
+            bindings = getAppsBindings(state);
+        } else if (props.post.channel_id === getThreadAppsBindingsChannelID(state)) {
```

Comment:
The different casing of `ChannelId` and `ChannelID` looks weird here. There was recently a discussion on this, siding for `ChannelId` https://community-daily.mattermost.com/core/pl/xpo386p6wprzx8xbtwy48mo46r